zebrad/src/commands/start.rs

```diff
@@ -285,6 +286,18 @@ impl StartCmd {
         info!("spawning syncer task");
         let syncer_task_handle = tokio::spawn(syncer.sync().in_current_span());
 
+        #[cfg(feature = "zebra-scan")]
+        // Spawn never ending scan task.
+        let scan_task_handle = {
+            info!("spawning zebra_scanner");
+            let mut storage = zebra_scan::storage::Storage::new();
+            for (key, birthday) in config.shielded_scan.sapling_keys_to_scan.iter() {
+                storage.add_sapling_key(key.clone(), Some(zebra_chain::block::Height(*birthday)));
+            }
+
+            tokio::spawn(zebra_scan::scan::start(state, storage).in_current_span())
+        };
```

Comment:
If you need a dummy handle, here's how to make one:
https://doc.rust-lang.org/std/future/fn.pending.html

And also remove the feature from every use of the task handle, because we will always have a task handle.

```suggestion
            tokio::spawn(zebra_scan::scan::start(state, storage).in_current_span())
        };

        #[cfg(not(feature = "zebra-scan"))]
        // Spawn a dummy scan task which doesn't do anything and never finishes.
        let scan_task_handle = tokio::spawn(std::future::pending().in_current_span());
```

Reply:
Yea, this is what is needed. Thanks!

https://github.com/ZcashFoundation/zebra/pull/7989/commits/cd8b0fd36845e1f96112bb95b6c6455c03a6a737